### PR TITLE
Krhubert/api client

### PR DESCRIPTION
**Pull request by NicolasMahe**

### CODE REVIEW

File: api/core/listen_result.go [hunk 1]
```diff
@@ -37,39 +37,38 @@ func (s *Server) ListenResult(request *ListenResultRequest, stream Core_ListenRe
...
+		return nil
 	}
 	_, ok := service.Tasks[taskKey]
 	if ok {
-		return
+		return nil
 	}
-	err = errors.New("Task '" + taskKey + "' doesn't exist in this service")
-	return
+	return fmt.Errorf("Task '%q' doesn't exist in this service", taskKey)
```
Comment (ilgooz) [inline api/core/listen_result.go]: remove ' around %q. %q already puts double quotes around the world.

File: api/core/listen_result.go [hunk 2]
```diff
@@ -37,39 +37,38 @@ func (s *Server) ListenResult(request *ListenResultRequest, stream Core_ListenRe
...
 	if taskKey == "" {
-		err = errors.New("Cannot filter output without specifying a task")
-		return
+		return fmt.Errorf("Cannot filter output without specifying a task")
 	}
 	task, ok := service.Tasks[taskKey]
 	if !ok {
-		err = errors.New("Task '" + taskKey + "' doesn't exist in this service")
-		return
+		return fmt.Errorf("Task '%q' doesn't exist in this service", taskKey)
```
Comment (ilgooz) [inline api/core/listen_result.go]: remove ' around %q. %q already puts double quotes around the world.

File: api/core/listen_result.go [hunk 3]
```diff
@@ -37,39 +37,38 @@ func (s *Server) ListenResult(request *ListenResultRequest, stream Core_ListenRe
...
 	}
+
+	var err error
 	_, ok = task.Outputs[outputFilter]
 	if !ok {
-		err = errors.New("Output '" + outputFilter + "' doesn't exist in the task '" + taskKey + "' of this service")
+		err = fmt.Errorf("Output %q doesn't exist in the task %q of this service", outputFilter, taskKey)
```
Comment (ilgooz) [inline api/core/listen_result.go]: return err here directly

File: api/core/listen_result.go [hunk 4]
```diff
@@ -37,39 +37,38 @@ func (s *Server) ListenResult(request *ListenResultRequest, stream Core_ListenRe
...
 	}
-	return
+	return err
```
Comment (ilgooz) [inline api/core/listen_result.go]: return nil

File: api/core/listen_result.go (hunk 1 above)
Comment (NicolasMahe) [reply api/core/listen_result.go]: Done

File: api/core/listen_result.go (hunk 2 above)
Comment (NicolasMahe) [reply api/core/listen_result.go]: Done

File: api/core/listen_result.go (hunk 3 above)
Comment (NicolasMahe) [reply api/core/listen_result.go]: I remove the `if taskKey == ""` test and return the error `Output %q doesn't exist in the....` directly

File: api/core/listen_result.go (hunk 4 above)
Comment (NicolasMahe) [reply api/core/listen_result.go]: Done

**Pull request merged**